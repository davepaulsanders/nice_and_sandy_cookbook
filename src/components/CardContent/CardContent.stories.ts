import type { Meta, StoryObj } from '@storybook/react';
 
import CardContent from './CardContent';
 
const meta: Meta<typeof CardContent> = {
  component: CardContent,
};
 
export default meta;
type Story = StoryObj<typeof CardContent>;
export const Card: Story = {args: {cardContent: {label: "TikTok Pasta", href: "https://www.delish.com/cooking/recipe-ideas/a35421563/baked-feta-pasta-tiktok/"}}}
