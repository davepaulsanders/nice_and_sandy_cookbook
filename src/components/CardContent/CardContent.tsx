import Actions from "../Actions/Actions";
import { Recipe } from "../../types/types";
const CardContent = ({ label, href, ..._rest }: Recipe) => {
	return (
		<div className="m-2">
			<Actions label={label} href={href} type="link" />
		</div>
	);
};
export default CardContent;
