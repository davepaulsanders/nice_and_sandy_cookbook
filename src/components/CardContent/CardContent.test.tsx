import CardContent from "./CardContent" 
import { expect, test } from 'vitest'
import {render } from '@testing-library/react'
import '@testing-library/dom'

test('<CardContent /> renders', () => {
	const { container } = render(<CardContent label="test" href="www.google.com"/>)
	expect(container).toMatchSnapshot()
})
