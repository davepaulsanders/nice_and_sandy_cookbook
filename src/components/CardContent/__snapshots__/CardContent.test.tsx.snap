// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<CardContent /> renders 1`] = `
<div>
  <div
    class="m-2"
  >
    <p
      class="text-lightg text-md"
    >
      Vegan Fajitas 1
    </p>
  </div>
</div>
`;
